tests/functional/test_dependencies.py

```diff
@@ -95,6 +96,34 @@ def test_dependency_using_reference(ref, recwarn, dependency_manager):
     assert DeprecationWarning not in [w.category for w in recwarn.list]
 
 
+def test_get_npm_version_from_spec():
+    assert get_npm_version_from_spec("1.0") == "==1.0.0"
+    assert get_npm_version_from_spec("1") == "==1.0.0"
+    assert get_npm_version_from_spec("1.0.0-beta") == "==1.0.0-beta"
+    assert get_npm_version_from_spec("1.0-beta") == "==1.0.0-beta"
+    assert get_npm_version_from_spec("1-beta") == "==1.0.0-beta"
+    assert get_npm_version_from_spec("1.0.0-beta.1") == "==1.0.0-beta.1"
+    assert get_npm_version_from_spec("1.0-beta.1") == "==1.0.0-beta.1"
+    assert get_npm_version_from_spec("1-beta.1") == "==1.0.0-beta.1"
+
+    assert get_npm_version_from_spec("~=1.0") == "~=1.0"
+    assert get_npm_version_from_spec(">=1") == ">=1.0"
+    assert get_npm_version_from_spec("<=1.0.0-beta") == "<=1.0.0-beta"
+    assert get_npm_version_from_spec(">1.0-beta") == ">1.0.0-beta"
+    assert get_npm_version_from_spec("<1-beta") == "<1.0.0-beta"
```

Comment:
Rather than have the cases with spaces (omg that rhymed) on the right side of the assertions in the test below, could we include them in this test here?